Release: 2.0.5 (#6069)
Kong/kong: CHANGELOG.md

- [0.9.9 and prior](#099---20170202)

## [2.0.5]

> Released 2020/06/30

### Dependencies

> - Bump OpenSSL version from `1.1.1f` to `1.1.1.g`.

Reviewer: ```suggestion
- Bump OpenSSL version from `1.1.1f` to `1.1.1g`.
```

  [#5820](https://github.com/Kong/kong/pull/5810)
- Bump [go-pluginserver](https://github.com/Kong/go-pluginserver) from version
  from `0.2.0` to `0.3.2`, leveraging [go-pdk](https://github.com/Kong/go-pdk) `0.3.1`.

>   See go-pdk changelog [here](https://github.com/Kong/go-pdk/blob/master/CHANGELOG.md#v031).

Reviewer: ```suggestion
  See [go-pdk changelog](https://github.com/Kong/go-pdk/blob/master/CHANGELOG.md#v031).
```

### Fixes

##### Core

> - Fix a race condition leading to random TLS breakages in DB-less mode.

Reviewer: ```suggestion
- Fix a race condition leading to random config fetching failure in DB-less mode.
```

  [#5833](https://github.com/Kong/kong/pull/5833)

> - Fix issue where a respawned worker would not get the existing configuration

Reviewer: ```suggestion
- Fix issue where a respawned worker would not use the existing configuration
```

  [#5873](https://github.com/Kong/kong/pull/5873)
- Go: fix issue with Go plugins where the plugin instance would be
  intermittently killed.
  Thanks [primableatom](https://github.com/primableatom) for the patch!
  [#5903](https://github.com/Kong/kong/pull/5903)
- Auto-convert `config.anonymous` from empty string to the `ngx.null` value.
  [#5906](https://github.com/Kong/kong/pull/5906)
- Fix issue where DB-less wouldn't correctly validate input with missing IDs,
  names, or cache key.
  [#5929](https://github.com/Kong/kong/pull/5929)
- Fix issue where a request to the upstream health endpoint would fail with
  HTTP 500 Internal Server Error.
  [#5943](https://github.com/Kong/kong/pull/5943)
- Fix issue where providing a declarative configuration file containing
  fields with explicit null values would result in an error.
  [#5999](https://github.com/Kong/kong/pull/5999)
- Fix issue where the balancer wouldn't be built for all workers.
  [#5931](https://github.com/Kong/kong/pull/5931)
- Fix issue where a declarative configuration file with primary keys specified
  as numbers would result in an error.
  [#6005](https://github.com/Kong/kong/pull/6005)

> - Fix 

Reviewer: ```suggestion
```

##### CLI

##### Configuration

- Fix issue where the Postgres password from the Kong confiuration file
  would be truncated if it contained a `#` character.
  [#5822](https://github.com/Kong/kong/pull/5822)

##### Admin API

- Fix issue where a `PUT` request on `/upstreams/:upstreams/targets/:targets`
  would result in HTTP 500 Internal Server Error.
  [#6012](https://github.com/Kong/kong/pull/6012)

##### PDK

- Stop request processing flow if body encoding fails.
  [#5829](https://github.com/Kong/kong/pull/5829)
- Ensure `kong.service.set_target()` includes the port number if a non-default
  port is used.
  [#5996](https://github.com/Kong/kong/pull/5996)

##### Plugins

- Go: fix issue where the go-pluginserver would not reload Go plugins'
  configurations.
  Thanks [wopol](https://github.com/wopol) for the patch!
  [#5866](https://github.com/Kong/kong/pull/5866)
- basic-auth: avoid fetching credentials when password is not given.
  Thanks [Abhishekvrshny](https://github.com/Abhishekvrshny) for the patch!
  [#5880](https://github.com/Kong/kong/pull/5880)

> - cors: avoid overwriting upstream response `Very` header; new values are now

Reviewer 2: Very = Vary
